### PR TITLE
feature/StringConcatenator

### PR DESCRIPTION
**Pull request by davidhopkinson26**
Added transfoormer StringConcatenator to strings.py.  It concatenates input coolumns into a single string output column.

Added test module and example notebook.

### CODE REVIEW

**Review by TommyMatthews — COMMENTED** [review]
This feels a bit vague - not immediately clear that values can be any datatype, maybe:

Transform to combine data from specified columns, of mixed datatypes, into a new column containing one string? 

(will this work on e.g. a column of dictionaries?)

File: tubular/strings.py [hunk 1]
```diff
@@ -135,3 +135,56 @@ def transform(self, X):
...
+        self.new_column = new_column
+
+        if not isinstance(separator, str):
+            raise TypeError(f"{self.classname()}: The separator should be a str")
+
+        self.separator = separator
+
+    def transform(self, X):
+        """
+        Concatenates values into one string.
```
Comment (TommyMatthews) [inline tubular/strings.py:170]: Again these descriptions feel a bit vague

File: examples/strings/StringConcatenator.ipynb [hunk 2]
```diff
@@ -0,0 +1,294 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# StringConcatenator\n",
+    "This notebook shows the functionality of the `StringConcatenator` class. This transformer joins values from different columns into a new string in a new column."
```
Comment (TommyMatthews) [inline examples/strings/StringConcatenator.ipynb:8]: same again here if we're changing values here to something more specific

File: tubular/strings.py [hunk 3]
```diff
@@ -135,3 +135,56 @@ def transform(self, X):
         )
 
         return X
+
+
+class StringConcatenator(BaseTransformer):
+    """
+    Transformer to concatenate values from a few columns into 1 string.
```
Comment (TommyMatthews) [inline tubular/strings.py:142]: Sorry the first comment (repeated below) should have been about the "concatenate values from a few columns" in this docstring - all the others reference this with "again"

This feels a bit vague - not immediately clear that values can be any datatype, maybe:

Transform to combine data from specified columns, of mixed datatypes, into a new column containing one string?

(will this work on e.g. a column of dictionaries?)

File: tubular/strings.py (hunk 1 above)
Comment (davidhopkinson26) [reply tubular/strings.py]: [7ade9ba](https://github.com/lvgig/tubular/pull/52/commits/7ade9ba12014db183180f7b0450691ed7867c468)

File: tubular/strings.py (hunk 3 above)
Comment (davidhopkinson26) [reply tubular/strings.py]: [7ade9ba](https://github.com/lvgig/tubular/pull/52/commits/7ade9ba12014db183180f7b0450691ed7867c468)

File: examples/strings/StringConcatenator.ipynb [hunk 4]
```diff
@@ -1,11 +1,12 @@
 {
  "cells": [
   {
+   "attachments": {},
    "cell_type": "markdown",
    "metadata": {},
    "source": [
     "# StringConcatenator\n",
-    "This notebook shows the functionality of the `StringConcatenator` class. This transformer joins values from different columns into a new string in a new column."
+    "This notebook shows the functionality of the `StringConcatenator` class. Thjis Transformer combines data from specified columns, of mixed datatypes, into a new column containing one string."
```
Comment (TommyMatthews) [inline examples/strings/StringConcatenator.ipynb:9]: typo

Comment (davidhopkinson26) [reply examples/strings/StringConcatenator.ipynb]: [bd7793b](https://github.com/lvgig/tubular/pull/52/commits/bd7793bcdc4cd14dd8ac29ccceb3a036b7037efd)

**Review by TommyMatthews — APPROVED** [review]

**Pull request merged**